File: src/j/implementation/microSet/InnerSet_0.java
package j.implementation.microSet;


import javax.swing.*;
import java.util.Collection;
import java.util.Iterator;

public class InnerSet_0 <T> extends AbstractInnerSet<T>
{
	public static InnerSet_0<Object> singleton = new InnerSet_0<>();
	
	@Override
	public InnerSet<T> addElement(T p_element)
	{
		return new InnerSet_1<T>(p_element);
	}

	@Override
	public InnerSet<T> addUnChecked(T p_element) {
		return addElement(p_element);
	}

	public InnerSet<T> addAllElements(InnerSet<T> innerSet) { return innerSet; }

	@Override
	public InnerSet<T> removeElement(Object p_element)
	{
		return this;
	}

	@Override
	public InnerSet<T> removeAllElements(InnerSet<T> innerSet) {
		return this;
	}

	@Override
	public InnerSet<T> retainAllElements(InnerSet<T> innerSet) {
		return this;
	}

	@Override
	public InnerSet<T> addAndPropagate(T p_element, MicroSet<T> microSetPropagate) {
		microSetPropagate.add(p_element);
		return new InnerSet_1<>(p_element);
	}

	@Override
	public InnerSet<T> addAllAndPropagate(InnerSet<T> innerSetToAdd, MicroSet<T> microSetPropagate) {
		microSetPropagate.innerSet.addAllElements(innerSetToAdd);
		return innerSetToAdd;
	}

	public T getElement(int index)
	{
		return null;
	}
	
	@Override
	public String toString()
	{
		return "{}";
	}
	
	@Override
	public int getSize()
	{
		return 0;
	}

	@Override
	public Iterator<T> iterator() {
		return new InnerSetIterator<>(this);
	}

	@Override
	public InnerSet<T> copy() {
		return (InnerSet<T>) InnerSet_0.singleton;
	}

	@Override
	public InnerSet<T> clear(boolean unused) {
		return this;
	}

	@Override
	public boolean containsElement(Object p_element)
	{
		return false;
	}

	@Override
	public boolean containsAllElements(InnerSet<T> innerSet) {
		return false;
	}
}

File: src/cpp/implementation/generate/inner_set_0.cpp
#include "inner_set_0.hpp"

template<typename T>
inner_set_0<T>::inner_set_0()
{

}

template<typename T>
inner_set_0<T>::~inner_set_0()
{

}

template<typename T>
inner_set<T>* inner_set_0<T>::add_element(T p_element)
{
	inner_set_1<T>* r_i1 = new inner_set_1<T>(p_element);

	return r_i1;

	//return nullptr;
}

template<typename T>
inner_set<T>* inner_set_0<T>::add_elements(T p_elements)
{
	return nullptr;
}

template<typename T>
bool inner_set_0<T>::contains_element(T p_element)
{
    return false;
}

template<typename T>
bool inner_set_0<T>::contains_all_elements(T p_elements)
{
    return false;
}

template<typename T>
T inner_set_0<T>::get_element(int p_index)
{
    return (T) NULL;
}

template<typename T>
inner_set<T>* inner_set_0<T>::remove_element(T p_element)
{
    return this;
}

template<typename T>
inner_set<T> inner_set_0<T>::retain_all_element(T p_elements)
{
    return *this;
}

template<typename T>
int inner_set_0<T>::get_size()
{
    return 0;
}

int main()
{
	inner_set_0<int> i0;

	return 0;
}
